### PR TITLE
Add missing components and clean up logging format

### PR DESCRIPTION
**Pull request by proferabg**

### CODE REVIEW

File: protocolize-api/src/main/java/dev/simplix/protocolize/api/item/Attribute.java [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+import lombok.NoArgsConstructor;
+
+import java.util.UUID;
+
+@Data
+@AllArgsConstructor
+@NoArgsConstructor
+public class Attribute {
+
+    AttributeType type;
```
Comment (Exceptionflug) [inline protocolize-api/src/main/java/dev/simplix/protocolize/api/item/Attribute.java:15]: Is there a reason why those fields are not private?

File: protocolize-api/src/main/java/dev/simplix/protocolize/api/item/BannerLayer.java [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+
+import lombok.AllArgsConstructor;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+
+@Data
+@AllArgsConstructor
+@NoArgsConstructor
+public class BannerLayer {
+    boolean direct;
```
Comment (Exceptionflug) [inline protocolize-api/src/main/java/dev/simplix/protocolize/api/item/BannerLayer.java:11]: Same here

File: protocolize-api/src/main/java/dev/simplix/protocolize/api/item/Bee.java [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+import lombok.AllArgsConstructor;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+import net.querz.nbt.tag.CompoundTag;
+
+@Data
+@AllArgsConstructor
+@NoArgsConstructor
+public class Bee {
+    CompoundTag entityData;
```
Comment (Exceptionflug) [inline protocolize-api/src/main/java/dev/simplix/protocolize/api/item/Bee.java:12]: And here

File: protocolize-api/src/main/java/dev/simplix/protocolize/api/item/BlockPredicate.java [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+import net.querz.nbt.tag.CompoundTag;
+
+import java.util.List;
+
+@Data
+@AllArgsConstructor
+@NoArgsConstructor
+public class BlockPredicate {
+
+    BlockSet blockSet;
```
Comment (Exceptionflug) [inline protocolize-api/src/main/java/dev/simplix/protocolize/api/item/BlockPredicate.java:15]: And here

File: protocolize-api/src/main/java/dev/simplix/protocolize/api/item/BlockSet.java [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+import lombok.Data;
+import lombok.NoArgsConstructor;
+
+import java.util.List;
+
+@Data
+@AllArgsConstructor
+@NoArgsConstructor
+public class BlockSet {
+    Integer type;
```
Comment (Exceptionflug) [inline protocolize-api/src/main/java/dev/simplix/protocolize/api/item/BlockSet.java:13]: And even here

File: protocolize-api/src/main/java/dev/simplix/protocolize/api/item/Book.java [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+import lombok.Data;
+import lombok.NoArgsConstructor;
+
+import java.util.List;
+
+@Data
+@AllArgsConstructor
+@NoArgsConstructor
+public class Book {
+    String title;
```
Comment (Exceptionflug) [inline protocolize-api/src/main/java/dev/simplix/protocolize/api/item/Book.java:14]: Also here

File: protocolize-api/src/main/java/dev/simplix/protocolize/api/item/Firework.java [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+import lombok.NoArgsConstructor;
+
+import java.util.List;
+
+@Data
+@AllArgsConstructor
+@NoArgsConstructor
+public class Firework {
+
+    int flightDuration;
```
Comment (Exceptionflug) [inline protocolize-api/src/main/java/dev/simplix/protocolize/api/item/Firework.java:14]: And here

File: protocolize-api/src/main/java/dev/simplix/protocolize/api/item/Firework.java [hunk 8]
```diff
@@ -0,0 +1,36 @@
...
+    List<Meta> explosions;
+
+    @Data
+    @AllArgsConstructor
+    @NoArgsConstructor
+    public static class Meta {
+        Shape shape;
```
Comment (Exceptionflug) [inline protocolize-api/src/main/java/dev/simplix/protocolize/api/item/Firework.java:21]: Even here

File: protocolize-api/src/main/java/dev/simplix/protocolize/api/item/SoundEvent.java [hunk 9]
```diff
@@ -0,0 +1,16 @@
...
+import dev.simplix.protocolize.data.Sound;
+import lombok.AllArgsConstructor;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+
+@Data
+@AllArgsConstructor
+@NoArgsConstructor
+public class SoundEvent {
+    Sound sound;
```
Comment (Exceptionflug) [inline protocolize-api/src/main/java/dev/simplix/protocolize/api/item/SoundEvent.java:13]: Also here

File: protocolize-api/src/main/java/dev/simplix/protocolize/api/item/Trim.java [hunk 10]
```diff
@@ -0,0 +1,41 @@
...
+import lombok.NoArgsConstructor;
+
+import java.util.Map;
+
+@Data
+@AllArgsConstructor
+@NoArgsConstructor
+public class Trim {
+
+    Integer materialType;
```
Comment (Exceptionflug) [inline protocolize-api/src/main/java/dev/simplix/protocolize/api/item/Trim.java:15]: And there

File: protocolize-api/src/main/java/dev/simplix/protocolize/api/item/Trim.java [hunk 11]
```diff
@@ -0,0 +1,41 @@
...
+    TrimMaterial trimMaterial;
+    Integer patternType;
+    TrimPattern trimPattern;
+
+    @Data
+    @AllArgsConstructor
+    @NoArgsConstructor
+    public static class TrimMaterial {
+        String assetName;
```
Comment (Exceptionflug) [inline protocolize-api/src/main/java/dev/simplix/protocolize/api/item/Trim.java:24]: Here also

File: protocolize-api/src/main/java/dev/simplix/protocolize/api/item/Trim.java [hunk 12]
```diff
@@ -0,0 +1,41 @@
...
+        float itemModelIndex;
+        Map<Integer, String> overrides;
+        ChatElement<?> description;
+    }
+
+    @Data
+    @AllArgsConstructor
+    @NoArgsConstructor
+    public static class TrimPattern {
+        String assetName;
```
Comment (Exceptionflug) [inline protocolize-api/src/main/java/dev/simplix/protocolize/api/item/Trim.java:35]: And here

**Review by Exceptionflug — CHANGES_REQUESTED** [review]

File: protocolize-api/src/main/java/dev/simplix/protocolize/api/item/Trim.java (hunk 12 above)
Comment (proferabg) [reply protocolize-api/src/main/java/dev/simplix/protocolize/api/item/Trim.java]: fixed

File: protocolize-api/src/main/java/dev/simplix/protocolize/api/item/Trim.java (hunk 11 above)
Comment (proferabg) [reply protocolize-api/src/main/java/dev/simplix/protocolize/api/item/Trim.java]: fixed

File: protocolize-api/src/main/java/dev/simplix/protocolize/api/item/Trim.java (hunk 10 above)
Comment (proferabg) [reply protocolize-api/src/main/java/dev/simplix/protocolize/api/item/Trim.java]: fixed

File: protocolize-api/src/main/java/dev/simplix/protocolize/api/item/SoundEvent.java (hunk 9 above)
Comment (proferabg) [reply protocolize-api/src/main/java/dev/simplix/protocolize/api/item/SoundEvent.java]: fixed

File: protocolize-api/src/main/java/dev/simplix/protocolize/api/item/Firework.java (hunk 8 above)
Comment (proferabg) [reply protocolize-api/src/main/java/dev/simplix/protocolize/api/item/Firework.java]: fixed

File: protocolize-api/src/main/java/dev/simplix/protocolize/api/item/Book.java (hunk 6 above)
Comment (proferabg) [reply protocolize-api/src/main/java/dev/simplix/protocolize/api/item/Book.java]: fixed

File: protocolize-api/src/main/java/dev/simplix/protocolize/api/item/Firework.java (hunk 7 above)
Comment (proferabg) [reply protocolize-api/src/main/java/dev/simplix/protocolize/api/item/Firework.java]: fixed

File: protocolize-api/src/main/java/dev/simplix/protocolize/api/item/BlockSet.java (hunk 5 above)
Comment (proferabg) [reply protocolize-api/src/main/java/dev/simplix/protocolize/api/item/BlockSet.java]: fixed

File: protocolize-api/src/main/java/dev/simplix/protocolize/api/item/BlockPredicate.java (hunk 4 above)
Comment (proferabg) [reply protocolize-api/src/main/java/dev/simplix/protocolize/api/item/BlockPredicate.java]: fixed

File: protocolize-api/src/main/java/dev/simplix/protocolize/api/item/Bee.java (hunk 3 above)
Comment (proferabg) [reply protocolize-api/src/main/java/dev/simplix/protocolize/api/item/Bee.java]: fixed

File: protocolize-api/src/main/java/dev/simplix/protocolize/api/item/BannerLayer.java (hunk 2 above)
Comment (proferabg) [reply protocolize-api/src/main/java/dev/simplix/protocolize/api/item/BannerLayer.java]: fixed

File: protocolize-api/src/main/java/dev/simplix/protocolize/api/item/Attribute.java (hunk 1 above)
Comment (proferabg) [reply protocolize-api/src/main/java/dev/simplix/protocolize/api/item/Attribute.java]: fixed

**Pull request merged**